cmake/prefix.cmake

```diff
@@ -20,9 +20,7 @@ if (NOT DEV_BUILD AND
     ((CMAKE_C_COMPILER_ID MATCHES "Clang" AND (CMAKE_C_COMPILER_VERSION VERSION_GREATER "3.8" OR CMAKE_C_COMPILER_VERSION VERSION_EQUAL "3.8")) OR
     (CMAKE_C_COMPILER_ID STREQUAL "GNU" AND (CMAKE_C_COMPILER_VERSION VERSION_GREATER "4.4.7" OR CMAKE_C_COMPILER_VERSION VERSION_EQUAL "4.4.7"))))
     # Clang integrated assembler does not support this option.
-    set(CMAKE_REQUIRED_FLAGS "-Wa,--debug-prefix-map")
-    check_c_source_compiles("int main(void) {}" HAVE_AS_DEBUG_PREFIX_MAP)
-    unset(CMAKE_REQUIRED_FLAGS)
+    check_c_compiler_flag(-Wa,--debug-prefix-map== HAVE_AS_DEBUG_PREFIX_MAP)
```

Comment:
Looks like this check fails on both clang and gcc:

```
$ clang -Wa --debug-prefix-map==
clang: error: unsupported option '--debug-prefix-map=='; did you mean '-fdebug-prefix-map=='?
clang: error: no input files
$ gcc -Wa --debug-prefix-map==
gcc: error: unrecognized command line option ‘-Wa’; did you mean ‘-W’?
gcc: fatal error: no input files
compilation terminated.
```

Reply:
Looks like you forgot the comma after `-Wa,`

Reply:
Looks like after a comma one can put anything:

```
$ gcc -Wa,--foobar==
gcc: fatal error: no input files
compilation terminated.
$ clang -Wa,--foobar==
clang: error: no input files
```

Reply:
Looks like this option is for assembler and is only recognized at link time.

Reply:
I see. Missed that. Looks okay then.